WordPress/gutenberg
# Align naming modals

<img width="1026" alt="Screenshot 2024-06-24 at 13 25 27" src="https://github.com/WordPress/gutenberg/assets/846565/3d2b1c6e-4335-4c47-88cf-7507699fe308">


## What?
Apply consistent design and behavior to naming modals in the site editor. Specifically:

* Utilise one of the `size` options for `Modal` (most commonly `small`).
* Auto-focus the name input when the modal is opened.

## Why?
Consistency is good. Some modals (e.g. renaming a block) already exhibit these features.

## Testing Instructions
In the following flows, renaming should invoke a modal with a specific `size`, and the name input should be auto-focussed:

- Rename pattern from command palette
- Rename navigation menu from dark sidebar
- Rename entity from data view 🗒️ 
- Rename entity from Inspector 🗒️ 
- Rename pattern category
- Create new template from the Inspector
- Create Pattern from data view

🗒️  In these scenarios all action modals will receive the given props. There may be a better way to do this, but for now it could be okay given renaming is the only action that invokes a modal.
